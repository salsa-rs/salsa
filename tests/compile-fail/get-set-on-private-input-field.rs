mod a {
    #[salsa::input]
    pub struct MyInput {
        field: u32,
    }
}

fn main() {
    let mut db = salsa::DatabaseImpl::new();
    let input = a::MyInput::new(&mut db, 22);

    input.field(&db);
    input.set_field(&mut db).to(23);
}
