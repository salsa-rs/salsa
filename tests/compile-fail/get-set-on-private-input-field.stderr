error[E0624]: method `field` is private
  --> tests/compile-fail/get-set-on-private-input-field.rs:12:11
   |
2  |     #[salsa::input]
   |     --------------- private method defined here
...
12 |     input.field(&db);
   |           ^^^^^ private method

error[E0624]: method `set_field` is private
  --> tests/compile-fail/get-set-on-private-input-field.rs:13:11
   |
2  |     #[salsa::input]
   |     --------------- private method defined here
...
13 |     input.set_field(&mut db).to(23);
   |           ^^^^^^^^^ private method
